### PR TITLE
main: add grpc-gateway support

### PR DESCRIPTION
**Pull request by xiang90**
Now etcd can serve HTTP json request at /v3alpha/

/cc @heyitsanthony @philips

### CODE REVIEW

File: cmd/Godeps/Godeps.json [hunk 1]
```diff
@@ -70,21 +70,37 @@
...
+			"Rev": "dcb844349dc5d2cb0300fdc4d2d374839d0d2e13"
+		},
+		{
 			"ImportPath": "github.com/ghodss/yaml",
 			"Rev": "73d445a93680fa1a78ae23a5839bad48f32ba1ee"
 		},
 		{
 			"ImportPath": "github.com/gogo/protobuf/proto",
-			"Comment": "v0.2-13-gc3995ae",
+			"Comment": "v0.1-163-gc3995ae",
```
Comment (gyuho) [inline cmd/Godeps/Godeps.json]: do we downgrade `gogo/protobuf`?

Comment (xiang90) [inline cmd/Godeps/Godeps.json]: this is just comment. no idea why it changed.

Comment (gyuho) [inline cmd/Godeps/Godeps.json]: Yeah godep sometimes does that... No other way to manually revert that part :0

**Comment by heyitsanthony** [comment]
lgtm

**Pull request merged**